dataset/src/main/scala/frameless/ops/NaFunctions.scala

```diff
@@ -0,0 +1,53 @@
+package frameless
+package ops
+
+import shapeless.ops.hlist.ToTraversable
+import shapeless.{HList, SingletonProductArgs}
+import syntax._
+
+sealed abstract class DropStrategy(val value: String)
+
+object DropStrategy {
+  /** Drops rows only if every specified column is null or NaN for that row. */
+  case object All extends DropStrategy("all")
+
+  /** Drops rows containing any null or NaN values. */
+  case object Any extends DropStrategy("any")
+}
+
+class NaFunctions[T](self: TypedDataset[T])(implicit val encoder: TypedEncoder[T]) {
```

Comment:
I think for Frameless it make sense to have dropping NaN to only work for Numeric (gives more type-safety). If you are trying to drop NaN on a string column, probably means you are doing something wrong and these kind or errors the compiler can help you with.

Similarly, you want drop None to work/compile only for `Option[_]` columns. Also, if you drop all None from a column, it means that you want to now remove the `Option` from the schema. That's what #307 PR is doing.

Reply:
That makes sense @imarios , I'm not aware of #307. Let me edit my PR.